igvc_gazebo/nodes/sim_color_detector/main.cpp

```diff
@@ -0,0 +1,95 @@
+#include <ros/ros.h>
+#include <sensor_msgs/Image.h>
+#include <sensor_msgs/image_encodings.h>
+#include <cv_bridge/cv_bridge.h>
+#include <opencv2/opencv.hpp>
+#include <opencv2/core/mat.hpp>
+#include <fstream>
+#include <iostream>
+#include <igvc_utils/NodeUtils.hpp>
+
+ros::Publisher img_pub;
+
+void handle_image(const sensor_msgs::ImageConstPtr& msg) {
+
+  cv_bridge::CvImagePtr cv_ptr;
+  cv::Mat frame;
+
+  try {
+    cv_ptr = cv_bridge::toCvCopy(msg, "bgr8");
+  } catch (cv_bridge::Exception& e) {
+    ROS_ERROR("CV-Bridge error: %s", e.what());
+    return;
+  }
+
+  frame = cv_ptr->image;
+  cv::Mat output(frame.rows, frame.cols, CV_8UC1, cv::Scalar::all(0));
+
+  const int min_b_line = 120;
+  const int max_b_line = 140;
+  const int min_g_line = 0;
+  const int max_g_line = 20;
+  const int min_r_line = 70;
+  const int max_r_line = 85;
+
+  const int max_b_obstacle_black = 45;
```

Comment:
What are black, orange, and white supposed to represent here? Add a comment describing what the colors mean: 
```
/**
max_*_obstacle_black -> black portion of the barricade
max_*_obstacle_orange -> orange portion of the barricade
min_*_obstacle_white -> white portion of the barricade 
*/
```